services/src/modules/directives/policy/policy.ts

```diff
@@ -15,11 +15,17 @@ export class PolicyDirective extends SchemaDirectiveVisitor {
       context: RequestContext,
       info: GraphQLResolveInfo
     ) => {
-      if (!context.ignorePolicies) {
+      if (!context.ignorePolicies && !policy.postValidate) {
         await context.policyExecutor.validatePolicy(policy, source, {}, context, info);
       }
 
-      return originalResolveObject ? await originalResolveObject(source, fields, context, info) : source;
+      const result = originalResolveObject ? await originalResolveObject(source, fields, context, info) : source;
+
+      if (!context.ignorePolicies && policy.postValidate) {
+        await context.policyExecutor.validatePolicy(policy, source, {}, context, info, result);
+      }
```

Comment:
Random thought, but since `ignorePolicies` is on the context and we pass the context to `validatePolicy`, maybe we should stop checking `if (!context.ignorePolicies)` here and just call `validatePolicy` anyway, having `validatePolicy` itself not do anything if `ignorePolicies` is set

It will save checking this in quite a few places